lib/ansible/galaxy/collection/__init__.py

```diff
@@ -218,6 +218,13 @@ def build_collection(collection_path, output_path, force):
     :return: The path to the collection build artifact.
     """
     b_collection_path = to_bytes(collection_path, errors='surrogate_or_strict')
+
+    b_galaxy_yml = os.path.join(b_collection_path, b'galaxy.yml')
+    if not os.path.exists(b_galaxy_yml):
```

Comment:
I don't think this check belongs here: it's already implemented in `_get_meta_from_src_dir()`. If we want to modify the message, we could just use try-except or modify the message in that function.

Reply:
I think the error there may generally be helpful - it complains about deps not being found. But for building the collection, that would probably be confusing since the problem is really that the metadata doesn't exist rather than an issue about dependencies. We could use a try-except here though.

Reply:
I've tweaked it locally but haven't pushed yet